### DISCUSSION TITLE
Auto Bed Leveling UBL not reaching bed

### DISCUSSION THREAD

**Discussion by Sonicboom247** [Jyers/Marlin#1784] [Q&A]
ok i give up, i am sure this is an easy fix but it's not clicking in my head, when I home with BL Touch it reached all 4 corners perfectly after  i set the Z offset to 5.75 even though the nozzle is still about 3-4mm away from bed.  Then i go to Create new mesh and the Z move up about 20-30mm above bed and then probe deploys and about 5-7mm above bed it stops and goes back up.  now if i put my finger under the probe it succeeds and goes to next probe point. if i leave it alone it just sits there indefinitely trying to probe the same point it can't reach... please  help

**Comment by Sonicboom247** [comment]
https://user-images.githubusercontent.com/84334417/149663692-c9010a5e-744c-4681-baf1-695816f4bd69.mp4

**Comment by BlueWings172** [comment]
@Sonicboom247 

Which version of the the firmware are you using ? is it precompiled ?

I noticed in your LCD screen that your z offset was set to -6 which is much bigger than it should. I remember I used to have the same BL Touch mount as you and my Z Offset was less than -3 (-2.75 i think but no where around -4 or -6). isn't it this one ?  https://www.thingiverse.com/thing:4584071/files

The latest firmware seems to have an issue. I had experienced some hair-pulling moments myself recently.

**Comment by Sonicboom247** [comment]
Hi and thanks for replying, it started off as Jyers Mesh No probe, but then i remembered i had a probe so added it to the Ender as it was the most straight forward connection.  So i enabled everything i needed for the probe to that same firmware but although it did home and test and do everything correctly it seemed to think my Z home was 20mm or so higher.  i then tested the E3V2-UBL-BLTouch-10x10-HS-v4.2.2-v2.0.1 but my extruder would not move, homing was perfect though...lol so i went back to my original as i could still print with that one.  I tried -6, -3, 0, 6 and all do the same.  it's as if though my Z MIN is 20 somewhere

**Comment by BlueWings172** [comment]
@Sonicboom247 

So have you already tried version 2.02h in this post #1694 ?

**Reply by Sonicboom247** [reply]
E3V2-UBL-BLTouch-10x10-v4.2.2-v2.0.2h is definitely doing what it is supposed to so far! just a few things i would like to edit/change for my own personal like just have show the Printer name vs Ender-3V2 E3V2-UBL-BLTouch-10x10-v4.2.2-v2.0.2h ready, auto init eeprom and a few other tweaks

**Comment by Sonicboom247** [comment]
all except the extruder does not extrude anything so now i have a nice gouge on a new bed were filament should have been

**Reply by BlueWings172** [reply]
From the LCD, if you go to the move menu and ask the extruder to extrude say 100mm what does it say or do ?

**Reply by Sonicboom247** [reply]
If i tell it to extrude manually it extrudes. Same happened last night with another pre compiled firmware. I believe another user also posted the same issue.

**Reply by Sonicboom247** [reply]
[Ender T_Bar_handle.zip](https://github.com/Jyers/Marlin/files/7877947/Ender.T_Bar_handle.zip)
this is a pretty simple one i was testing with

**Reply by Sonicboom247** [reply]
Thank you! I had deleted that from my setup but kept using he same file... Shame on me.

**Reply by Sonicboom247** [reply]
and for the previews, yes i fixed that as well last night but again kept using the same darn file..
![20220116_172907](https://user-images.githubusercontent.com/84334417/149680811-86609fe4-e6cc-4d56-a837-260b60af62db.jpg)
testing with a surfboard now

**Reply by Sonicboom247** [reply]
thanks so much again! hopefully other can learn from my mistakes.  i set a super wide offset just to be safe, and of course it didn't stick to the bed but at least it was extruding, i wonder why it didn't just pause the print for heat up or error out.. oh well. moving on and running through some fresh pid tunes and nice warmed up bed levels to run a real print shortly.  maybe tomorrow i will try to tweak the firmware so i can just show the printer name and stuff.  might have to edit background if all previews will be dark like that. i am guessing it is because it is set to black PLA +

**Reply by Sonicboom247** [reply]
would this be a better /more efficient Start G-Code? what i want to achieve is
Bed preheat for ABL
Extruder preheat for print but not enough to ooze
ABL go through calibration/verification
Hand off heating to gcode to begin printing

***PreHeat***
M190 S60 ; preheat bed for BLTouch and wait
M104 S150 ; Preheat Extruder but not too hot no need to wait
***Auto Bed Leveling***
G28 ; Auto Home
G29 A ; Activate Bed Leveling
G29 L0 ; Load mesh slot 0
G29 J2 ; Probe Bed for mesh tilting
***Priming***
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z up to avoid scratching bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y150.0 Z0.3 F1500 E20 ; Draw first line
G1 X0.4 Y150.0 Z0.3 F5000 ; Move to side
G1 X0.4 Y20 Z0.3 F1500.0 E15 ; Draw second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z up to prevent scratching bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish

**Reply by K2Kevin** [reply]
Do you really need to level each time when you are using Mesh?  I never have...

I use:
M420 S1 ; Load Mesh


My Start:

; Ender 3 Custom Start G-code
G92 E0 ; Reset Extruder
G28 ; Home all axes
M420 S1 ; Load Mesh
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish

**Reply by Sonicboom247** [reply]
> Do you really need to level each time when you are using Mesh? I never have...

Haha, on my 3 other printers, Artillery 3D, I Manual bed level maybe once every few months.  With this new holy grail of bed leveling i would like to see if there is a difference, i would probably love to see an option for after (x) number of prints, automatically check mesh.

I will give yours a shot!

**Reply by K2Kevin** [reply]
FYI, I do 5x5 Mesh, not UBL.

**Reply by Sonicboom247** [reply]
so looks like M420 S1 is the same as G29 A but for UBL

**Reply by Sonicboom247** [reply]
> FYI, I do 5x5 Mesh, not UBL.

yup, on my no probe printers i do same 5x5, this is my first time with a probe.

**Reply by K2Kevin** [reply]
I thought G29 is just single use before print normal bed leveling (Not mesh).  Advantage of Mesh was it is saved on board so you don't have to run auto level for each print.

**Reply by Sonicboom247** [reply]
i'm really just wanting to get the nozzle to heat up to about 150 before the 4 point probe it does after mesh creation.  So if i am understanding it correctly, with UBL it takes about 10 min to probe almost 100 points on the bed to make a mesh, then once that mesh is saved, it can do a quick 3 or 4 point verification and quick adjustment just before printing.

Right now i can get it to heat up the bed to 50 but then it seems to ignore the M104 S150 and goes straight to build temp of 212.

**Reply by Sonicboom247** [reply]
G29 - Bed Leveling (Unified)
Requires AUTO_BED_LEVELING_UBL.
G28 disables bed leveling. Follow with G29 A to turn leveling on, or use RESTORE_LEVELING_AFTER_G28 to automatically keep leveling on after G28.
M420 S1 can be used to turn leveling on, but requires a valid (complete) mesh. See M420 for more details.

**Reply by K2Kevin** [reply]
I must be missing something.  You set the temp in Cura Profile (Initial Temp Settings). My printer doesn't move until it reaches those temps, then does it homing and starts the print. I assume that doesn't work for you?

**Reply by Sonicboom247** [reply]
you are correct, you and I set the print job temps in Cura, i am trying to get the start Gcode ***Before*** the print job so ABL can do its thing but Cura is saying "nope i'll take it from here"...lol pretty sure i am missing a simple command

**Reply by Sonicboom247** [reply]
yup, looks like i should have an M192 instead of M104

**Reply by Sonicboom247** [reply]
@tititopher68-dev , thank you again, I think.  Please understand that with the BLTouch I am a beginner, I am not sure if you are scolding us or some of the instruction is lost in translation.  
"You lose it because your information is far from always accurate" this is how troubleshooting works, we bounce ideas off each other to better understand so we can then single in on the correct answer, sometimes we learn other things from it.  as you can see @K2Kevin may have learned about UBL vs ABL. My information was posted straight from https://marlinfw.org/docs/gcode/G029-ubl.html.

"although a priori you persist with your temperatures and always reversing..."  did you mean Prior or Priority? and by persist, do you mean I am supposed to do temperatures first or that i am always reversing my commands? 

"It is important (I have already advised you) to make your prior settings (Z-Offset and Mesh at printing temperature)." Yes, this is exactly what I want but at printing temperatures my extruder was oozing so when the print started there was no filament in the nozzle. I therefore wanted to either add a retract before UBL and reset extruder or lower temp on nozzle.  however now that I understand nozzle expansion, i know i want it at print temp. so thank you for that knowledge. 

either way, thank you

**Reply by K2Kevin** [reply]
@Sonicboom247 couldn't agree with you more.  Might be a surprise to some, but I completely understand the concepts behind ABL Mesh and UBL Mesh and the differences between them.  Since I do not use UBL, Im not aware of any special commands, or its own idiosyncrasies.  

What I have learned, we are dealing with, at times, very buggy software and hardware that is less than optimal.  Often, processes do not work or perform as expected or intended.  Just because a process has been working flawless for me, doesn't mean it will work at all for you, due to some bug related to software, hardware or moon phase.  I continue to have my own strange findings working with ABL, Mesh, Z-Stop and Manual bed leveling processes.  With very little documentation, we find ourselves asking "dumb" questions, and often the answers can make even less sense in a logical world.  <grin>

Hope you get a handle on what you are trying to accomplish and squash a few more bugs in the meantime.

**Reply by Sonicboom247** [reply]
@tititopher68-dev  Thank you again, i did get your start gcode from the zip file and the printer just started successfully printing with mesh and all, just a little oozing still but as i do not know how to start the calibration without filament engaged, i think a simple retract before Mesh would be the easiest.  I do have a soft wire brush for cleaning nozzle so that is covered.

please feel free to reach out to me for any testing you might need as i have other printers so i can have one down for testing if needed. I am a beginner to the BL Touch and UBL and with only 1 year into 3D printing, heck i still might be a beginner. but agree with you about people having their answers quicker by reading a few posts., In the past i ran a saltwater reef forum and the main thing we focused on having are stickies.  Topics that contain valuable information stuck to the top of the forum for users to read first.  Nowadays the answer needs to be voted up but you still have to weed through everything to find your answer to your specific setup and specific issue.

**Reply by K2Kevin** [reply]
@tititopher68-dev funny, I just had this convo with a client last week as we discussed 3D printing one of his prototypes/Molds.  He wanted to get a 3D printer in house and do it himself, rather than sending it off to another company for print (I DO NOT 3d Print for clients).  I basically told him, it's not like purchasing an HP or Epson.  NOTHING is plug and play, as the conversation turned to, remembering IRQ's, Com Ports, Dip switches etc.  

"Even some of the best works began with many failed prints."
